backend/thirdparty/linking.go

```diff
@@ -49,19 +46,42 @@ func LinkAccount(tx *pop.Connection, cfg *config.Config, p persistence.Persister
 }
 
 func link(tx *pop.Connection, cfg *config.Config, p persistence.Persister, userData *UserData, providerName string, user *models.User, isSaml bool) (*AccountLinkingResult, error) {
-	if !isSaml && !cfg.ThirdParty.Providers.Get(providerName).AllowLinking {
-		return nil, ErrorUserConflict("third party account linking for existing user with same email disallowed")
+	if !isSaml {
+		if strings.HasPrefix(providerName, "custom_") {
+			provider, ok := cfg.ThirdParty.CustomProviders[strings.TrimPrefix(providerName, "custom_")]
```

Comment:
Couldn't `thirdparty.GetProvider()` provide the `AllowLinking` attribute. I think that would be more readable.